monai/utils/misc.py

```diff
@@ -217,3 +218,39 @@ def set_determinism(
         torch.backends.cudnn.benchmark = False
     else:
         torch.backends.cudnn.deterministic = False
+
+
+def list_to_dict(items):
+    """
+    To convert a list of "key=value" pairs into a dictionary.
+    For examples: items: `["a=1", "b=2", "c=3"]`, return: {"a": "1", "b": "2", "c": "3"}.
+    If no "=" in the pair, use None as the value, for example: ["a"], return: {"a": None}.
+
+    """
+
+    def _parse_var(s):
+        items = s.split("=")
+        # we remove blanks around keys, as is logical
+        key = items[0].strip()
+        value = None
+        if len(items) > 1:
+            # rejoin the rest, for example: MODEL="checkpoint_best_metric=0.94.pth"
+            value = "=".join(items[1:])
+        return key, value
+
+    d = dict()
+    if items:
+        for item in items:
+            key, value = _parse_var(item)
+
+            try:
```

Comment:
I would suggest `ast.literal_eval` would be a more robust solution.

Reply:
Hi @ericspod ,

I tried it and got some errors:
```
>>> print(literal_eval("false"))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/conda/lib/python3.6/ast.py", line 85, in literal_eval
    return _convert(node_or_string)
  File "/opt/conda/lib/python3.6/ast.py", line 84, in _convert
    raise ValueError('malformed node or string: ' + repr(node))
ValueError: malformed node or string: <_ast.Name object at 0x7f3820bed7b8>
>>> print(literal_eval("haha"))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/conda/lib/python3.6/ast.py", line 85, in literal_eval
    return _convert(node_or_string)
  File "/opt/conda/lib/python3.6/ast.py", line 84, in _convert
    raise ValueError('malformed node or string: ' + repr(node))
ValueError: malformed node or string: <_ast.Name object at 0x7f3820c0ad68>
```
Should we use `eval`?

Thanks.

Reply:
`literal_eval` will parse only literal values in a safe way, so something like "x=3" won't parse. You can use `literal_eval` and resort to keeping the original string if this exception is raised rather than having the nested try/except blocks.

Reply:
But we also need to parse bool type as shown in the above test example.
So I still use the previous try-catch for bool and string and use `literal_eval ` for int and float?

Thanks.

Reply:
updated in the latest commit.
Thanks.